File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch10/C/P10-qGauss.cpp
// Gaussian quadratures
#include <stdio.h>
#include <math.h>
#include "integral.h"

double Func1(double x) { return 12*pow(x,11); }
double Func2(double x) { return pow(x,5) * exp(-x); }

int main()
{
   int n, nmax;

   nmax = 6;
   
   printf("Integral of 12*x^11 on [-1,1]\n");
   for (n=2; n<=nmax; n++) {
      printf("n = %i  I1 = %18.14g\n",n,qGaussLeg(Func1,0e0,1e0,n));
   }
   printf("\nIntegral of x^5 * exp(-x) on [0,+inf)\n");
   for (n=2; n<=nmax; n++) {
      printf("n = %i  I2 = %18.14g\n",n,qGaussLag(Func2,0e0,n));
   }
}

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch10/Python/P10-qGauss.py
# Gaussian quadratures
from math import *
from integral import *

def Func1(x): return 12*pow(x,11)
def Func2(x): return x*x*x * exp(-x)

# main

nmax = 6
x = [0]*(nmax+1)
w = [0]*(nmax+1)

print("Integral of 12*x^11 on [-1,1]")
for n in range(2,nmax+1):
   print("n = ",n,"I1 = ",qGaussLeg(Func1,0e0,1e0,n))

print("\nIntegral of x^3 * exp(-x) on [0,+inf)")
for n in range(2,nmax+1):
   print("n = ",n,"I2 = ",qGaussLag(Func2,0e0,n))
